rules/internal/objc_provider_utils.bzl

```diff
@@ -37,7 +37,7 @@ def _merge_objc_providers(providers, transitive = []):
     )
     return apple_common.new_objc_provider(**objc_provider_fields)
 
-def _merge_dynamic_framework_providers(dynamic_framework_providers):
+def _merge_dynamic_framework_providers(dynamic_framework_providers, supports_cc_info_in_dynamic_framework_provider):
```

Comment:
nit: lets just check for bazel 7 here directly like we do elsewhere, it makes it easier to remove in the future than delete the param